include/Spectra/DavidsonSymEigsSolver.h

```diff
@@ -39,8 +39,8 @@ class DavidsonSymEigsSolver : public JDSymEigsBase<DavidsonSymEigsSolver<OpType>
     Vector m_diagonal;
 
 public:
-    DavidsonSymEigsSolver(OpType& op, Index nev, Index nvec_init, Index nvec_max) :
-        JDSymEigsBase<DavidsonSymEigsSolver<OpType>, OpType>(op, nev, nvec_init, nvec_max)
+    DavidsonSymEigsSolver(OpType& op, Index nev, Index nvec_init, Index nvec_max, std::unique_ptr<LoggerBase<Scalar, Vector>> logger = nullptr) :
```

Comment:
I do not prefer default arguments instead just have two constructors. The reason for it is that if we eve change the interface or add a second default argument it will all break down in user code.